### DISCUSSION TITLE
My autocommand doesn't work? How to even check that?

### DISCUSSION THREAD

**Discussion by marcinjahn** [LazyVim/LazyVim#3870] [Q&A] [answered]
I've modified the `$HOME/.config/nvim/lua/config/autocmds.lua` with the following content:

```lua
-- Autocmds are automatically loaded on the VeryLazy event
-- Default autocmds that are always set: https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/autocmds.lua
-- Add any additional autocmds here

local function augroup(name)
  return vim.api.nvim_create_augroup("lazyvim_" .. name, { clear = true })
end

vim.api.nvim_create_autocmd({ "FileType" }, {
  group = augroup("my_roslyn"),
  pattern = "*.cs",
  callback = function(ev)
    print("Does it even work?")
    vim.lsp.start({
      name = "my-roslyn",
      cmd = { "/home/mnj/code/private/roslyn-ls-wrapper/bin/Debug/net8.0/roslyn-ls-wrapper" },
      root_dir = vim.fs.root(ev.buf, { "*.sln" }),
      filetypes = { "cs" },
    })
  end,
})
```

I'd expect the log to be printed (via noice) and LSP to get started whenever I open a csharp (*.cs) file. However, nothing happens when I open such a file. Am I doing something wrong?

Also, what is the proper way to set up my own LSP? Should I create a plugin for it? Or is the autocmd approach alright?

**Comment by folke** [comment, marked as answer]
The pattern for a `FileType` event should be the filetype, not that glob.
Alternatively, you can use that glob expression, but with a `BufRead` event.

But you likely want the FT, so just use the correct filetype as a pattern

**Reply by marcinjahn** [reply]
Thanks! The log appears, now I'm one step closer to actually having the LSP started.
